### ISSUE TITLE
[rush] project dependency tree error with Create React App

### ISSUE THREAD

**Issue by semopz** [microsoft/rushstack#2069]
**Is this a feature or a bug?**

- [ ] Feature
- [x] Bug

**Please describe the actual behavior.**
I'm trying to create a mono repo that contains SPFx (SharePoint Framework) 1.4.1 webparts and Create React Apps. As soon as I add the SPFx webpart as a package in `rush.json` and run `rush update`, I can no longer run `npm start` in the CRA. Note that the error message refers to webpack but I've also seen Jest as the offending package.

```
npm start   

> cra-app@0.1.0 start C:\dev\github\cra-spfx\cra\cra-app
> react-scripts start


There might be a problem with the project dependency tree.
It is likely not a bug in Create React App, but something you need to fix locally.

The react-scripts package provided by Create React App requires a dependency:

  "webpack": "4.42.0"

Don't try to install it manually: your package manager does it automatically.
However, a different version of webpack was detected higher up in the tree:

  C:\dev\github\cra-spfx\common\temp\node_modules\.pnpm\node_modules\webpack (version: 3.6.0)

Manually installing incompatible versions is known to cause hard-to-debug issues.

If you would prefer to ignore this check, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That will permanently disable this message but you might encounter other issues.

To fix the dependency tree, try following the steps below in the exact order:

  1. Delete package-lock.json (not package.json!) and/or yarn.lock in your project folder.
  2. Delete node_modules in your project folder.
  3. Remove "webpack" from dependencies and/or devDependencies in the package.json file in your project folder.
  4. Run npm install or yarn, depending on the package manager you use.

In most cases, this should be enough to fix the problem.
If this has not helped, there are a few other things you can try:

  5. If you used npm, install yarn (http://yarnpkg.com/) and repeat the above steps with it instead.
     This may help because npm has known issues with package hoisting which may get resolved in future versions.

  6. Check if C:\dev\github\cra-spfx\common\temp\node_modules\.pnpm\node_modules\webpack is outside your project directory.
     For example, you might have accidentally installed something in your home folder.

  7. Try running npm ls webpack in your project folder.
     This will tell you which other package (apart from the expected react-scripts) installed webpack.

If nothing else helps, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That would permanently disable this preflight check in case you want to proceed anyway.

P.S. We know this message is long but please read the steps above :-) We hope you find them helpful!

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! cra-app@0.1.0 start: 'react-scripts start'
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the cra-app@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     \Roaming\npm-cache\_logs\2020-08-03T19_45_07_541Z-debug.log
```

**If the issue is a bug, how can we reproduce it? Please provide detailed steps and include a GitHub branch if applicable. Your issue will get resolved faster if you can make it easy to investigate.**
1. Clone https://github.com/semopz/cra-spfx.git
2. `rush install`
3. cd \cra\cra-app
4. `npm start`
5. observe error
6. comment out the wp-2019 package in `rush.json`
7. `rush update`
8. This time `npm start` in the CRA will work fine

**What is the expected behavior?**
I'd like to be able to add projects to the monorepo that have dependencies on outdated packages alongside other projects that use more recent versions of those packages. I don't think there's a supported way to update packages that SPFx uses so I'm trying to bundle everything up in a CRA where I can use more up to date stuff.

**If this is a bug, please provide the tool version, Node.js version, and OS.**

* **Tool: Rush** 
* **Tool Version: 5.30.1** 
* **Node Version: 10.22.0** 
  * **Is this a LTS version? Yes** 
  * **Have you tested on a LTS version?** 
* **OS: W10 19041.388**

**Comment by iclanton** [comment]
This appears to be a bug in `create-react-app`. CRA's `verifyPackageTree` script crawls up the filesystem looking for `node_modules` folders at every folder level, starting from the installed location of the `react-scripts` package in [this `while` loop](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/scripts/utils/verifyPackageTree.js#L54). The [comment](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/scripts/utils/verifyPackageTree.js#L51) in that code mentions that it's looking to ensure that there aren't other versions higher in the tree, which shouldn't be a check it needs to perform: correctly resolving to its own declared dependency is enough.

From the error message, it looks like CRA has never been tested with PNPM. I also tested this with PNPM workspaces on your branch, and that also doesn't work. I was able to work around this on your branch, however, by dropping a `"webpack": "4.42.0"` line in your repo's `common/config/rush/common-versions.json`'s `"preferredVersions"` field and running `rush update --full`. This won't work with PNPM workspaces, however.

So, in summary, CRA is performing a check that it probably shouldn't be. You can work around this by telling Rush that your preferred version of webpack (which will cause that version to end up in the `common/temp/node_modules` root) is the one that CRA wants.

**Comment by semopz** [comment]
Very helpful, thanks. This probably isn't an issue for Rush to fix so I'll close this
